editor/inserter/test/menu.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { shallow } from 'enzyme';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType, unregisterBlockType, getBlockTypes } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { InserterMenu } from '../menu';
+
+const textBlock = {
+	name: 'core/text-block',
+	title: 'Text',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const advancedTextBlock = {
+	name: 'core/advanced-text-block',
+	title: 'Advanced Text',
+	save: noop,
+	edit: noop,
+	category: 'common',
+};
+
+const moreBlock = {
+	name: 'core/more-block',
+	title: 'More',
+	save: noop,
+	edit: noop,
+	category: 'layout',
+	useOnce: 'true',
+};
+
+const youtubeBlock = {
+	name: 'core/youtube-block',
+	title: 'Youtube',
+	save: noop,
+	edit: noop,
+	category: 'embed',
+};
+
+describe( 'InserterMenu', () => {
+	afterEach( () => {
+		getBlockTypes().forEach( ( block ) => {
+			unregisterBlockType( block.name );
+		} );
+	} );
+
+	beforeEach( () => {
+		registerBlockType( textBlock.name, textBlock );
+		registerBlockType( advancedTextBlock.name, advancedTextBlock );
+		registerBlockType( moreBlock.name, moreBlock );
+		registerBlockType( youtubeBlock.name, youtubeBlock );
+	} );
+
+	it( 'Should show the recent tab by default', () => {
+		const wrapper = shallow(
+			<InserterMenu
+				instanceId={ 1 }
+				blocks={ [] }
+				recentlyUsedBlocks={ [] }
+			/>
+		);
+		const activeCategory = wrapper.find( '.editor-inserter__tab .is-active' );
+		const visibleBlocks = wrapper.find( '.editor-inserter__block' );
+		expect( activeCategory.text() ).toBe( 'Recent' );
+		expect( visibleBlocks.length ).toBe( 0 );
+	} );
+
+	it( 'Should show the recently used blocks in the recent tab', () => {
+		const wrapper = shallow(
+			<InserterMenu
+				instanceId={ 1 }
+				blocks={ [] }
+				recentlyUsedBlocks={ [ advancedTextBlock ] }
+			/>
+		);
+		const visibleBlocks = wrapper.find( '.editor-inserter__block' );
+		expect( visibleBlocks.length ).toBe( 1 );
```

Comment:
I'm finding when I run this test in isolation (adding `.only`), this line fails because it's expecting `36` as length... sounds like the default set of registered blocks? Is this bleeding over from another test? Should we unregister all blocks in a `beforeAll` block here?